lib/Map/Vector/Protomaps/ProtomapsGeojsonSource.ts

```diff
@@ -0,0 +1,319 @@
+    // Create circle with 10 pixel radius to pick features
+    const buffer = circle(
+      [CesiumMath.toDegrees(longitude), CesiumMath.toDegrees(latitude)],
+      10 * zoomMeters,
+      {
+        steps: 10,
+        units: "meters"
+      }
+    );
+
+    // Create wrappedBuffer with only positive coordinates - this is needed for features which overlap antemeridian
+    const wrappedBuffer = cloneDeep(buffer);
+    wrappedBuffer.geometry.coordinates.forEach((ring) =>
+      ring.forEach((point) => {
+        point[0] = point[0] < 0 ? point[0] + 360 : point[0];
+      })
+    );
+
+    const bufferBbox = bbox(buffer);
+
+    // Get array of all features
+    const geojsonFeatures = this.geojsonObject.features;
+
+    const pickedFeatures: Feature[] = [];
+
+    for (let index = 0; index < geojsonFeatures.length; index++) {
+      const feature = geojsonFeatures[index];
+      if (!feature.bbox) {
+        feature.bbox = bbox(feature);
+      }
+
+      // Filter by bounding box and then intersection with buffer (to minimize calls to booleanIntersects)
+      if (
+        Math.max(
+          feature.bbox[0],
+          // Wrap buffer bbox if necessary
+          feature.bbox[0] > 180 ? bufferBbox[0] + 360 : bufferBbox[0]
+        ) <=
+          Math.min(
+            feature.bbox[2],
+            // Wrap buffer bbox if necessary
+            feature.bbox[2] > 180 ? bufferBbox[2] + 360 : bufferBbox[2]
+          ) &&
+        Math.max(feature.bbox[1], bufferBbox[1]) <=
+          Math.min(feature.bbox[3], bufferBbox[3])
+      ) {
+        // If we have longitudes greater than 180 - used wrappedBuffer
+        if (feature.bbox[0] > 180 || feature.bbox[2] > 180) {
+          if (booleanIntersects(feature, wrappedBuffer))
+            pickedFeatures.push(feature);
+        } else if (booleanIntersects(feature, buffer))
+          pickedFeatures.push(feature);
+      }
+    }
+
+    // Convert pickedFeatures to ImageryLayerFeatureInfos
+    pickedFeatures.forEach((f) => {
+      const featureInfo = new ImageryLayerFeatureInfo();
+
+      featureInfo.data = f;
+      featureInfo.properties = Object.assign(
+        { [LAYER_NAME_PROP]: GEOJSON_SOURCE_LAYER_NAME },
+        f.properties ?? {}
+      );
+
+      if (
+        f.geometry.type === "Point" &&
+        typeof f.geometry.coordinates[0] === "number" &&
+        typeof f.geometry.coordinates[1] === "number"
+      ) {
+        featureInfo.position = Cartographic.fromDegrees(
+          f.geometry.coordinates[0],
+          f.geometry.coordinates[1]
+        );
+      }
+
+      featureInfo.configureDescriptionFromProperties(f.properties);
+      featureInfo.configureNameFromProperties(f.properties);
+
+      featureInfos.push(featureInfo);
+    });
+
+    return featureInfos;
+  }
+}
+
+export const geomTypeMap = (
+  type: string | null | undefined
+): GeomType | null => {
+  switch (type) {
+    case "Point":
+    case "MultiPoint":
```

Comment:
Does the input for this function ever come from the code updated in #7432, or is that a separate code path?

Reply:
Most likely yes, but depends on a few other factors

Reply:
Just merged your PR in

Reply:
I'm not sure if your yes means that the input comes from the other code or if it is a separate code path, but thank you for merging the PR.